File: Python/67.Add Binary.py
#返回一个数加一后的值(数可以任意大)
class Solution(object):
    def addBinary(self, a, b):
        l1 , l2 ,flag = len(a)-1 , len(b)-1 , 0
        result  = ''
        while l1>=0 or l2>=0 or flag!=0:
            sum = 0
            if l1>=0:
                sum += int(a[l1])
            if l2>=0:
                sum += int(b[l2])
            sum += flag
            flag = sum//2
            result += str(sum%2)
            l1 , l2 = l1-1 , l2-1
        return result[::-1]

s = Solution()
print(s.addBinary('11','1'))

File: C++/Math/67.Add Binary.cpp
/*
两个字符串表示两个二进制数，求二者的和。
*/

#include <iostream>
#include <cstring>
using namespace std;

class Solution {
public:
	string addBinary(string a, string b) {
        int i = a.size()-1, j = b.size()-1,flag = 0,sum=0;
        string result="";
        while(i>=0 || j>=0 || flag)
        {
        	sum = flag;
        	if(i>=0)	sum += (a[i--]-'0');
        	if(j>=0)	sum += (b[j--]-'0');
        	flag = sum/2;
        	result = char(sum%2+'0')+result;
        }
        return result;
    }
};
int main()
{
	Solution s;
	cout<<s.addBinary("11","1")<<endl;
}

